### PR TITLE
Finish EurekaContainerHealthService and AggregatingContainerHealthService implementation

### PR DESCRIPTION
**Pull request by tbak**
### Description of the Change

Finish EurekaContainerHealthService and AggregatingContainerHealthService implementation

### CODE REVIEW

File: titus-common/src/main/java/com/netflix/titus/common/util/rx/ReactorHeadTransformer.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+import org.reactivestreams.Publisher;
+import reactor.core.Disposable;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.FluxSink;
+
+/**
+ * An operator that combines snapshots state with hot updates. To prevent loss of
+ * any update for a given snapshot, the hot subscriber is subscribed first, and its
+ * values are buffered until the snapshot state is stream to the subscriber.
```
Comment (corindwyer) [inline titus-common/src/main/java/com/netflix/titus/common/util/rx/ReactorHeadTransformer.java]: stream -> streamed

File: titus-ext/eureka/src/test/java/com/netflix/titus/ext/eureka/containerhealth/EurekaContainerHealthServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                    assertContainerHealth(healthService.getHealthStatus(taskId1), taskId1, ContainerHealthState.Unhealthy);
+                    assertContainerHealthEvent(event, taskId1, ContainerHealthState.Unhealthy);
+                })
+
+                .thenCancel()
+                .verify(Duration.ofSeconds(5));
+    }
+
+    @Test
+    public void testEurekaStaleDataCleanup() {
```
Comment (corindwyer) [inline titus-ext/eureka/src/test/java/com/netflix/titus/ext/eureka/containerhealth/EurekaContainerHealthServiceTest.java]: Add test for when eureka status goes from `UP` to `NOT_REGISTERED`

File: titus-common/src/main/java/com/netflix/titus/common/util/rx/ReactorHeadTransformer.java (hunk 1 above)
Comment (tbak) [reply titus-common/src/main/java/com/netflix/titus/common/util/rx/ReactorHeadTransformer.java]: Fixed

File: titus-ext/eureka/src/test/java/com/netflix/titus/ext/eureka/containerhealth/EurekaContainerHealthServiceTest.java (hunk 2 above)
Comment (tbak) [reply titus-ext/eureka/src/test/java/com/netflix/titus/ext/eureka/containerhealth/EurekaContainerHealthServiceTest.java]: Done

**Pull request merged**